From 4081b65897b91e94c2b5919e5a36ff03886b0b42 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Mon, 11 Nov 2013 12:00:42 +0100
Subject: [PATCH] [osx-maverick-fix][unsequenced-modifications] Removed the SQR
 macro

Fix following OSX 10.9 warning:
/oce/src/math/math_Powell.cxx:177:2: warning: multiple unsequenced modifications to 'sqrarg' [-Wunsequenced]
        SQR(PreviousMinimum-TheMinimum -del)-del*
        ^
/oce/src/math/math_Powell.cxx:33:23: note: expanded from macro 'SQR'
                      ^
---
 src/math/math_Powell.cxx | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

Index: oce/src/math/math_Powell.cxx
===================================================================
--- oce.orig/src/math/math_Powell.cxx
+++ oce/src/math/math_Powell.cxx
@@ -29,8 +29,7 @@
 #include <math_MultipleVarFunction.hxx>
 
 
-static Standard_Real sqrarg;
-#define SQR(a) (sqrarg=(a), sqrarg*sqrarg)
+
 
 class DirFunctionBis : public math_Function {
 
@@ -173,9 +172,10 @@ void math_Powell::Perform(math_MultipleV
     F.Value(ptt, fptt);
 
     if (fptt < PreviousMinimum) {
-      t = 2.0 *(PreviousMinimum -2.0*TheMinimum +fptt)*
-	SQR(PreviousMinimum-TheMinimum -del)-del*
-	  SQR(PreviousMinimum-fptt);
+      Standard_Real aux1 = PreviousMinimum - TheMinimum - del;
+      Standard_Real aux2 = PreviousMinimum - fptt;
+      t = 2.0 * (PreviousMinimum -2.0 * TheMinimum + fptt) * aux1 * aux1 -
+      del * aux2 * aux2;
       if (t <0.0) {
 	//Minimisation along the direction
 	Standard_Boolean IsGood = MinimizeDirection(TheLocation, xit,
